Docstring refresh

corinnebosley opened the pull request:
Closes #978 

Fixes for all warnings except HTTP errors 403 and 404.  Can't seem to get rid of them.

QuLogic commented:
The HTTP 404 is from the `reference_url` setting; change `'numpy': 'http://docs.scipy.org/doc/numpy/reference'` to `'numpy': 'https://docs.scipy.org/doc/numpy'` (also can add `https` to the rest).

The 403 is from shapely, but that seems to be some server configuration as the link works in a browser just fine.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -282,11 +284,9 @@ def __init__(self, access_token, map_id):
         Access to Mapbox web services requires an access token and a map ID.
         See https://www.mapbox.com/developers/api/ for details.
```
Can be updated to `https://www.mapbox.com/api-documentation/`

QuLogic commented on lib/cartopy/io/__init__.py:
```diff
@@ -100,29 +100,29 @@ class Downloader(object):
 
-        ``url_template`` - The template of the full URL representing this
-                           resource.
+        * ``url_template``: The template of the full URL representing this
```
I think these no longer need to be in code formatting (seems to apply to this entire file.)

corinnebosley replied:
Good spot.  Will fix this.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -131,14 +134,12 @@ def tile_bbox(self, x, y, z, y0_at_north_pole=True):
-        Parameters
-        ----------
-        x, y, z : int
+        :param x, y, z : int
```
Aren't you using Google-style docstrings all over?

corinnebosley replied:
Yeah, missed this one.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -412,16 +412,9 @@ def coastlines(self, resolution='110m', color='black', **kwargs):
-                           dataset. Currently can be one of "110m", "50m", and
-                           "10m".
-
-        .. note::
```
Where did this note go?

corinnebosley replied:
This was about something which was supposed to be fixed in v0.5, so it's either been fixed (which we're pretty sure it has, having spoken to @pp-mo about it), or it's just not worth pointing out any more.

corinnebosley commented:
Thanks @QuLogic, I appreciate the help with this.

QuLogic commented:
Wait, actually, the latest sphinx-gallery [no longer requires `reference_url` for third-party libraries](https://sphinx-gallery.github.io/advanced_configuration.html#link-to-documentation) when using intersphinx (which we do), so you can remove everything but the `cartopy` entry.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -96,8 +96,8 @@
-_MAP_SERVER_NS = '{http://mapserver.gis.umn.edu/mapserver}'
-_GML_NS = '{http://www.opengis.net/gml}'
+_MAP_SERVER_NS = '{https://mapserver.gis.umn.edu/mapserver}'
+_GML_NS = '{https://www.opengis.net/gml}'
```
XML namespaces should not be changed.

corinnebosley commented:
@pp-mo @QuLogic Can somebody please check this for me while I've got a big green tick, and merge it if you're happy.

pelson commented on docs/source/citation.rst:
```diff
@@ -16,7 +16,7 @@ For example::
  year = {2010 - 2015},
  address = {Exeter, Devon },
- url = {http://scitools.org.uk/cartopy}
+ url = {https://scitools.org.uk/cartopy}
```
This domain doesn't yet work 😢

pelson commented on docs/source/developer_interfaces.rst:
```diff
@@ -33,7 +33,7 @@ way.
 An example of specialising this class can be found in
 :mod:`cartopy.io.shapereader.NEShpDownloader` which enables the downloading of
-zipped shapefiles from the `<http://www.naturalearthdata.com>`_ website. All
+zipped shapefiles from the `<https://www.naturalearthdata.com>`_ website. All
```
Doesn't yet work with https

pelson commented on docs/source/matplotlib/advanced_plotting.rst:
```diff
@@ -5,7 +5,7 @@ From the outset, cartopy's purpose has been to simplify and improve the quality
 interface, in many cases the hardest part of producing such visualisations is getting
 hold of the data in the first place. To address this, a Python package,
-`Iris <http://scitools.org.uk/iris/>`_, has been created to make loading and saving data from a
+`Iris <https://scitools.org.uk/iris/>`_, has been created to make loading and saving data from a
```
👎

pelson commented on docs/source/tutorials/using_the_shapereader.rst:
```diff
@@ -23,7 +23,7 @@ Helper functions for shapefile acquisition
 Cartopy provides an interface for access to frequently used data such as the
 `GSHHS <https://www.ngdc.noaa.gov/mgg/shorelines/gshhs.html>`_ dataset and from
-the `NaturalEarthData <http://www.naturalearthdata.com/>`_ website. 
+the `NaturalEarthData <https://www.naturalearthdata.com/>`_ website.
```
👎

pelson commented on lib/cartopy/_crs.pyx:
```diff
@@ -133,16 +126,21 @@ cdef class CRS:
-        globe : :class:`~cartopy.crs.Globe` instance, optional
-            If omitted, the default Globe instance will be created.
-            See :class:`~cartopy.crs.Globe` for details.
+        Args:
```
Please target numpydoc format (https://numpydoc.readthedocs.io/en/latest/) wherever possible.

corinnebosley replied:
That was the form of the cartopy docs previously, but they weren't rendering properly with the use of sphinx-gallery.  I've spent a lot of hours changing them all to be not in this format, but more like the Iris docs.

corinnebosley commented:
I think these failing tests are probably license headers, so I'll deal with them in a minute.

Regarding the docs themselves, this is the best I can do to get them consistently using numpydoc convention.  They are still ugly so I am still unhappy, but they are better.  And all look the same.

corinnebosley commented:
Just in case I don't complete this today (as it's Friday afternoon and I'm leaving in 10 minutes); I think I might have found some more stuff to make the docs pretty:
https://github.com/numpy/numpy/blob/master/doc/source/conf.py#L87

corinnebosley commented:
Along with making sure all the tests pass for this PR, it looks like if I want to make the docs numpy-style-pretty, then I will need to add a scipy-sphinx-theme submodule to this repository.  I am unsure about this, and about whether it is a good idea or not, so I am going to make a new branch from this one where I can attempt it, see how the docs look and have a discussion about how to move forward.

corinnebosley commented:
I am hitting a problem in which the doctests fail with latest everything because of this difference:
Using numpy 1.9.3 and matplotlib 1.4.3:
```
    >>> import numpy as np
    >>> data = np.ones([5, 6]) * np.arange(6)
    >>> cyclic_data = add_cyclic_point(data)
    >>> print(cyclic_data)
[[ 0.  1.  2.  3.  4.  5.  0.]
 [ 0.  1.  2.  3.  4.  5.  0.]
 [ 0.  1.  2.  3.  4.  5.  0.]
 [ 0.  1.  2.  3.  4.  5.  0.]
 [ 0.  1.  2.  3.  4.  5.  0.]]
```

Using latest everything:
```
    >>> import numpy as np
    >>> data = np.ones([5, 6]) * np.arange(6)
    >>> cyclic_data = add_cyclic_point(data)
    >>> print(cyclic_data)
[[0. 1. 2. 3. 4. 5. 0.]
 [0. 1. 2. 3. 4. 5. 0.]
 [0. 1. 2. 3. 4. 5. 0.]
 [0. 1. 2. 3. 4. 5. 0.]
 [0. 1. 2. 3. 4. 5. 0.]]
```

I can remove some spaces from the result in the docstring, but then the doctest just fails in the other two Travis tests instead.  Does anybody know how I get around this?

corinnebosley commented:
More than that, does anybody know why this is happening now and wasn't before?

corinnebosley commented:
Never mind that last comment, it's because numpy have released 1.14 today (https://docs.scipy.org/doc/numpy-1.14.0/reference/) and array outputs have less spaces in now.

corinnebosley commented:
@pelson I am happy with this PR now, but I have made a lot of changes that I wasn't expecting to, so it's pretty big.

I changed the docstring style to numpydoc in a separate commit for each file (some of them are one commit for a few files, but only if there were hardly any changes per file).

I also had to make some changes following the release of numpy 1.14 today; the behavioural change in the release is to do with the number of spaces in an array repr, so the doctest for cartopy.util was failing for half of the test matrix.  I changed the expected result in the docstring to that which is produced by numpy 1.14, but this caused the other half of the test matrix (which uses numpy 1.9.3 and previous array formatting) to fail.  Subsequently I have changed the travis.yml so that it only runs doctests on 'Latest everything'.

I **really want to get these changes merged** so that I can close this monstrous PR, but there are more changes that I'd like to make afterwards, so I would like to discuss with you the following: 

In order to get prettier formatting (more like the actual numpy docs) in our documentation, we will need to use the package [scipy-sphinx-theme](https://github.com/scipy/scipy-sphinx-theme).  Numpy has used this as a submodule, but I feel that if we are going to use this package, then we should use it with the same settings across all our scitools repositories.  If this ends up being the case, then adding scipy-sphinx-theme as a submodule individually to all our repos seems inefficient, and maybe we should use it in a different way instead.

I'm going to open a new PR and play with scipy-sphinx-theme a bit, and hopefully we can find some time to make some decisions about this pretty soon.  Please let me know when you have time to talk about this.

QuLogic commented on lib/cartopy/io/__init__.py:
```diff
@@ -256,7 +252,7 @@ def acquire_resource(self, target_path, format_dict):
     def _urlopen(self, url):
         """
-        Return a file handle to the given HTTP resource URL.
+        Returns a file handle to the given HTTP resource URL.
```
Don't change.

QuLogic commented on lib/cartopy/io/__init__.py:
```diff
@@ -345,15 +338,17 @@ class RasterSource(object):
     def validate_projection(self, projection):
         """
-        Raise an error if this raster source cannot provide images in the
+        Raises an error if this raster source cannot provide images in the
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -49,39 +49,37 @@ def __init__(self, *args, **kwargs):
     def __getstate__(self):
         """
-        Override the default to ensure when pickling that any new attributes
+        Overrides the default to ensure when pickling that any new attributes
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -49,39 +49,37 @@ def __init__(self, *args, **kwargs):
     def bbox(self):
         """
-        Return a :class:`~shapely.geometry.polygon.Polygon` instance for
+        Returns a :class:`~shapely.geometry.polygon.Polygon` instance for
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -93,7 +91,7 @@ def bbox(self):
     def world_files(fname):
         """
-        Determine potential world filename combinations, without checking
+        Determines potential world filename combinations, without checking
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -161,7 +161,7 @@ def __array__(self):
     def from_world_file(cls, img_fname, world_fname):
         """
-        Return an Img instance from the given image filename and
+        Returns an Img instance from the given image filename and
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -175,7 +175,7 @@ def from_world_file(cls, img_fname, world_fname):
     def world_file_extent(worldfile_handle, im_shape):
         """
-        Return the extent ``(x0, x1, y0, y1)`` and pixel size
+        Returns the extent ``(x0, x1, y0, y1)`` and pixel size
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -228,24 +225,21 @@ def __init__(self, name, crs, images=None):
                           img_class=Img):
         """
-        Search the given directory for the associated world files
+        Searches the given directory for the associated world files
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -345,24 +337,27 @@ def _is_parent(parent, child):
     def image_for_domain(self, target_domain, target_z):
         """
-        Determine the image that provides complete coverage of target location.
+        Determines the image that provides complete coverage of target
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -450,17 +444,18 @@ def find_images(self, target_domain, target_z, start_tiles=None):
     def subtiles(self, collection_image):
         """
-        Find the higher resolution image tiles that compose this parent
+        Finds the higher resolution image tiles that compose this parent
```
Don't change.

QuLogic commented on lib/cartopy/io/img_nest.py:
```diff
@@ -472,25 +467,28 @@ def subtiles(self, collection_image):
     def get_image(self, collection_image):
         """
-        Retrieve the data of the target image from file.
+        Retrieves the data of the target image from file.
```
Don't change.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -133,15 +137,17 @@ def tile_bbox(self, x, y, z, y0_at_north_pole=True):
-            Whether the numbering of the y coordinate starts at the north
-            pole (as is the convention for Google tiles), or the south
-            pole (as is the convention for TMS).
+        x, y, z
```
Why remove the type?

corinnebosley replied:
Consistency.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -256,10 +262,11 @@ class StamenTerrain(GoogleTiles):
     Additional info:
-    http://mike.teczno.com/notes/osm-us-terrain-layer/background.html
-    http://maps.stamen.com/#terrain/12/37.6902/-122.3600
+    https://mike.teczno.com/notes/osm-us-terrain-layer/background.html
```
Doesn't support HTTPS.

corinnebosley replied:
Thank you, missed that.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -256,10 +262,11 @@ class StamenTerrain(GoogleTiles):
-    http://mike.teczno.com/notes/osm-us-terrain-layer/background.html
-    http://maps.stamen.com/#terrain/12/37.6902/-122.3600
+    https://mike.teczno.com/notes/osm-us-terrain-layer/background.html
+    https://maps.stamen.com/
```
Doesn't support HTTPS.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -277,18 +284,18 @@ class MapboxTiles(GoogleTiles):
     def __init__(self, access_token, map_id):
         """
-        Set up a new Mapbox tiles instance.
+        Sets up a new Mapbox tiles instance.
```
Don't change.

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -373,10 +380,11 @@ def tileextent(self, quadkey):
     def find_images(self, target_domain, target_z, start_tile=None):
         """
-        Find all the quadtree's at the given target zoom, in the given
+        Finds all the quadtrees at the given target zoom, in the given
```
Don't change (first word).

QuLogic commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -394,7 +402,7 @@ def find_images(self, target_domain, target_z, start_tile=None):
 
 def _merge_tiles(tiles):
-    """Return a single image, merging the given images."""
+    """Returns a single image, merging the given images."""
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -103,9 +104,11 @@
                           output_projection, output_extent, target_resolution):
     """
-    Reproject an Image from one source-projection and extent to another.
+    Reprojects an Image from one source-projection and extent to another.
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -151,7 +154,7 @@ def _warped_located_image(image, source_projection, source_extent,
 def _target_extents(extent, requested_projection, available_projection):
     """
-    Translate the requested extent in the display projection into a list of
+    Translates the requested extent in the display projection into a list of
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -267,7 +270,7 @@ def _native_srs(self, projection):
     def _fallback_proj_and_srs(self):
         """
-        Return a :class:`cartopy.crs.Projection` and corresponding
+        Returns a :class:`cartopy.crs.Projection` and corresponding
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -541,22 +544,26 @@ def _select_tiles(self, tile_matrix, tile_matrix_limits,
                      max_pixel_span):
         """
-        Add images from the specified WMTS layer and matrix set to cover
+        Adds images from the specified WMTS layer and matrix set to cover
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -680,7 +685,7 @@ def __init__(self, service, features, getfeature_extra_kwargs=None):
     def default_projection(self):
         """
-        Return a :class:`cartopy.crs.Projection` in which the WFS
+        Returns a :class:`cartopy.crs.Projection` in which the WFS
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -704,21 +709,22 @@ def default_projection(self):
     def fetch_geometries(self, projection, extent):
         """
-        Return any Point, Linestring or LinearRing geometries available
+        Returns any Point, Linestring or LinearRing geometries available
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -758,15 +764,17 @@ def fetch_geometries(self, projection, extent):
     def _to_shapely_geoms(self, response):
         """
-        Convert polygon coordinate strings in WFS response XML to Shapely
+        Converts polygon coordinate strings in WFS response XML to Shapely
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -809,20 +817,21 @@ def _to_shapely_geoms(self, response):
     def _find_polygon_coords(self, node, find_str):
         """
-        Return the x, y coordinate values for all the geometries in
+        Returns the x, y coordinate values for all the geometries in
```
Don't change.

QuLogic commented on lib/cartopy/io/ogc_clients.py:
```diff
@@ -857,7 +866,7 @@ def _find_polygon_coords(self, node, find_str):
     def _node_has_child(node, find_str):
         """
-        Return whether `node` contains (at any sub-level), a node with name
+        Returns whether `node` contains (at any sub-level), a node with name
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -92,7 +92,7 @@ def validate_projection(self, projection):
     def fetch_raster(self, projection, extent, target_resolution):
         """
-        Fetch SRTM elevation for the given projection and approximate extent.
+        Fetches SRTM elevation for the given projection and approximate extent.
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -125,7 +125,7 @@ def fetch_raster(self, projection, extent, target_resolution):
     def srtm_fname(self, lon, lat):
         """
-        Return the filename for the given lon/lat SRTM tile (downloading if
+        Returns the filename for the given lon/lat SRTM tile (downloading if
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -150,7 +150,7 @@ def srtm_fname(self, lon, lat):
     def combined(self, lon_min, lat_min, nx, ny):
         """
-        Return an image and its extent for the group of nx by ny tiles
+        Returns an image and its extent for the group of nx by ny tiles
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -238,7 +238,7 @@ def __init__(self, downloader=None, max_nx=3, max_ny=3):
 def srtm(lon, lat):
     """
-    Return (elevation, crs, extent) for the given longitude latitude.
+    Returns (elevation, crs, extent) for the given longitude latitude.
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -314,25 +320,25 @@ def srtm_composite(lon_min, lat_min, nx, ny):
 def read_SRTM(fh):
     """
-    Read the array of (y, x) elevation data from the given named file-handle.
+    Reads the array of (y, x) elevation data from the given named file-handle.
```
Don't change.

QuLogic commented on lib/cartopy/io/srtm.py:
```diff
@@ -369,7 +375,7 @@ def read_SRTM(fh):
 def SRTM3_retrieve(lon, lat):
     """
-    Return the path of a .hgt file for the given SRTM location.
+    Returns the path of a .hgt file for the given SRTM location.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/feature_artist.py:
```diff
@@ -33,7 +33,7 @@
 class _GeomKey(object):
     """
-    Provide id() based equality and hashing for geometries.
+    Provides id() based equality and hashing for geometries.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -95,12 +95,14 @@ class InterProjectionTransform(mtransforms.Transform):
     def __init__(self, source_projection, target_projection):
         """
-        Create the transform object from the given projections.
+        Creates the transform object from the given projections.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -244,13 +253,14 @@ class GeoAxes(matplotlib.axes.Axes):
     def __init__(self, *args, **kwargs):
         """
-        Create a GeoAxes object using standard matplotlib
+        Creates a GeoAxes object using standard matplotlib
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -307,15 +320,15 @@ def add_image(self, factory, *args, **kwargs):
     def hold_limits(self, hold=True):
         """
-        Keep track of the original view and data limits for the life of this
+        Keeps track of the original view and data limits for the life of this
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -534,29 +561,34 @@ def add_feature(self, feature, **kwargs):
     def add_geometries(self, geoms, crs, **kwargs):
         """
-        Add the given shapely geometries (in the given crs) to the axes.
+        Adds the given shapely geometries (in the given crs) to the axes.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -534,29 +561,34 @@ def add_feature(self, feature, **kwargs):
     def get_extent(self, crs=None):
         """
-        Get the extent (x0, x1, y0, y1) of the map in the given coordinate
+        Gets the extent (x0, x1, y0, y1) of the map in the given coordinate
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -617,12 +649,17 @@ def _get_extent_geom(self, crs=None):
     def set_extent(self, extents, crs=None):
         """
-        Set the extent (x0, x1, y0, y1) of the map in the given
+        Sets the extent (x0, x1, y0, y1) of the map in the given
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -667,10 +704,10 @@ def set_extent(self, extents, crs=None):
-        Set the extent of the Axes to the limits of the projection.
-
-        .. note::
+        Sets the extent of the Axes to the limits of the projection.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -697,26 +734,26 @@ def autoscale_view(self, tight=None, scalex=True, scaley=True):
-                    to another rectangular coordinate system are supported.
-
-        .. note::
+        Sets the x ticks.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -744,26 +781,26 @@ def set_xticks(self, ticks, minor=False, crs=None):
-                    to another rectangular coordinate system are supported.
-
-        .. note::
+        Sets the y ticks.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -791,7 +828,7 @@ def set_yticks(self, ticks, minor=False, crs=None):
     def stock_img(self, name='ne_shaded'):
         """
-        Add a standard image to the map.
+        Adds a standard image to the map.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -988,16 +1028,20 @@ def read_user_background_images(self, verify=True):
     def add_raster(self, raster_source, **slippy_image_kwargs):
         """
-        Add the given raster source to the GeoAxes.
+        Adds the given raster source to the GeoAxes.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1026,30 +1070,38 @@ def _regrid_shape_aspect(self, regrid_shape, target_extent):
     def imshow(self, img, *args, **kwargs):
         """
-        Add the "transform" keyword to :func:`~matplotlib.pyplot.imshow'.
+        Adds the "transform" keyword to :func:`~matplotlib.pyplot.imshow'.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1219,23 +1271,24 @@ def _boundary(self):
     def set_boundary(self, path, transform=None, use_as_clip_path=True):
         """
-        Given a path, update the :data:`.outline_patch` and
+        Given a path, updates the :data:`.outline_patch` and
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1588,11 +1645,12 @@ def _pcolormesh_patched(self, *args, **kwargs):
     def pcolor(self, *args, **kwargs):
         """
-        Add the "transform" keyword to :func:`~matplotlib.pyplot.pcolor'.
+        Adds the "transform" keyword to :func:`~matplotlib.pyplot.pcolor'.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1614,14 +1672,24 @@ def pcolor(self, *args, **kwargs):
-        Plot a field of arrows.
-
-        Extra Kwargs:
+        Plots a field of arrows.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1685,35 +1753,45 @@ def quiver(self, x, y, u, v, *args, **kwargs):
     def barbs(self, x, y, u, v, *args, **kwargs):
         """
-        Plot a field of barbs.
+        Plots a field of barbs.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1824,21 +1914,21 @@ def streamplot(self, x, y, u, v, **kwargs):
     def add_wmts(self, wmts, layer_name, wmts_kwargs=None, **kwargs):
         """
-        Add the specified WMTS layer to the axes.
+        Adds the specified WMTS layer to the axes.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/geoaxes.py:
```diff
@@ -1852,21 +1942,22 @@ def add_wmts(self, wmts, layer_name, wmts_kwargs=None, **kwargs):
     def add_wms(self, wms, layers, wms_kwargs=None, **kwargs):
         """
-        Add the specified WMS layer to the axes.
+        Adds the specified WMS layer to the axes.
```
Don't change.

QuLogic commented on lib/cartopy/mpl/gridliner.py:
```diff
@@ -35,7 +35,7 @@
 def _fix_lons(lons):
     """
-    Fix the given longitudes into the range ``[-180, 180]``.
+    Fixes the given longitudes into the range ``[-180, 180]``.
```
Don't change.

QuLogic commented on .gitignore:
```diff
@@ -1,54 +0,0 @@
-# standard python ignores
```
Why are you deleting this file?

corinnebosley replied:
That was an accident. Something to do with git clean and rebuilding docs I think.

corinnebosley commented on lib/cartopy/_crs.pyx:
```diff
@@ -115,7 +116,7 @@ class Globe(object):
     def to_proj4_params(self):
         """
-        Create an OrderedDict of key value pairs which represents this globe
+        Creates an OrderedDict of key value pairs which represents this globe
```
Look, there's been a lot of finnicky comments from @QuLogic and @pelson on this PR, but the one thing that I feel should be a priority is consistency in the documentation.  I changed this because some docstrings start with (for example) 'creates' and some with 'create'.  I don't particularly care which they are changed to, but they should be the same tense.

corinnebosley commented on lib/cartopy/data/shapefiles/gshhs/README.TXT:
```diff
@@ -1,139 +0,0 @@
-#	$Id: README.TXT,v 1.21 2011/07/01 20:48:23 guru Exp $
```
Accident.

corinnebosley commented:
@QuLogic I have tried to address all the docstring-tense issues there, but I still need to work out which things went missing (like the .gitignore and stuff) and get them back.  How am I doing so far?

corinnebosley commented:
@QuLogic I think that's your lot.  Let me know about any other changes you'd like to see.  I really appreciate the time you're putting into reviewing this.  Thanks.

QuLogic commented on lib/cartopy/mpl/ticker.py:
```diff
@@ -183,41 +182,39 @@ def __init__(self,
+        Parameters
+        ----------
+        zero_direction_label: optional
+            Boolean flag.
```
I don't think you added this to any other bool parameters?

corinnebosley commented on lib/cartopy/feature.py:
```diff
@@ -61,13 +61,18 @@ class Feature(six.with_metaclass(ABCMeta)):
-    Kwargs:
+    Other Parameters
+    ----------------
+    kwargs
```
I think if I add the stars to the beginning, it ends up looking like this when it's rendered:

kwargs(**)

corinnebosley replied:
Oh, no, my mistake, just tried it and it does get rendered appropriately.  I'll sort that out now.

corinnebosley commented on lib/cartopy/io/img_nest.py:
```diff
@@ -453,14 +447,15 @@ def subtiles(self, collection_image):
-            A tuple pair containing the parent
+        Parameters
+        ----------
+        collection_image: A tuple pair containing the parent
```
Yeah you're quite right.

corinnebosley commented on lib/cartopy/io/img_tiles.py:
```diff
@@ -280,15 +287,15 @@ def __init__(self, access_token, map_id):
-        map_id: str
-            A map ID for a publically accessible map. This is the map whose
-            tiles will be retrieved through this process.
+        zmap_id
```
I guess it must be.  Good spot.

corinnebosley commented:
@pelson I think I need a big green tick from you before anyone can merge this.  Not sure though.  Either way, whenever you have time I really really really want this merged.  Please.

pp-mo commented:
I've reviewed @pelson queries + I'm confident you've fixed all that stuff.

However, I'm still seeing other problems introduced by the transition to the numpy-docs format, 
e.g. like this one : 
![cartopy_projection_docs_prob](https://user-images.githubusercontent.com/2089069/34774826-d40ed76c-f608-11e7-88ff-5ed962d7bddb.png)

I don't know how much like this there is.
Probably not much at all, but if they are problems introduced by this change then I think it's still best they get fixed here, rather than merging + fixing subsequently.

Rather than request further changes from you (!), I think the best is if I do a quick independent run-through + propose some fixes as a PR to your branch.
Then when you're ok with that (+ merge it here), I'll be ok with the whole thing + will merge.

Watch this space ...

pp-mo commented:
I've reviewed @pelson queries + I'm confident you've fixed all that stuff.

However, I'm still seeing other problems introduced by the transition to the numpy-docs format, 
e.g. like this one : 
![rotatedpole_docs_prob](https://user-images.githubusercontent.com/2089069/34776102-8d47ce42-f60d-11e7-8759-6b6c5d8b98a0.png)

I don't know how much like this there is.
Probably not much at all, but if they are problems introduced by this change then I think it's still best they get fixed here, rather than merging + fixing subsequently.

Rather than request further changes from you (!), I think the best is if I do a quick independent run-through + propose some fixes as a PR to your branch.
Then when you're ok with that (+ merge it here), I'll be ok with the whole thing + will merge.

Watch this space ...

corinnebosley commented:
@pp-mo That's not what I see when I build the docs locally.  That looks more like how the current docs are presented.  I'm confused.

QuLogic commented:
Unfortunately, CircleCI does _not_ build a PR merge commit, so if I closed and opened this again, it would not build properly. If you merge `master` on your branch, it should start building properly here and we can be on the same page.

pelson commented on docs/source/matplotlib/advanced_plotting.rst:
```diff
@@ -54,6 +54,7 @@ Block plots
     :include-source:
 
     import iris
+    import iris_sample_data
```
That's an unused import... 😢

pelson replied:
(I know the *data* is used, but it is a wart to import something so that something else can do a thing that I'm really not keen on)

pelson commented:
🎉 🎆  @corinnebosley - great work!
@QuLogic - as ever, your contributions here have been invaluable! Thanks!

corinnebosley commented:
@pelson Thanks for merging that.  I'm not done yet though...

pelson commented:
> Thanks for merging that. I'm not done yet though...

Good. You are done with *this* PR though 😄 😉  (hint: make them smaller next time)
